### DISCUSSION TITLE
Control rules: Pump-Timeopen/Timeclosed

### DISCUSSION THREAD

**Discussion by yuhanmao1207** [pyswmm/pyswmm#449] [Q&A]
Hello, I am learning to use pyswmm to control the pump by open/close time. It's doable to use TIMEOPEN and TIMECLOSED rules in SWMM. I am wondering if there is a property for pumps called timeopen/timeclosed in pyswmm.

Thank you so much for taking time to read this!

**Comment by bemcdonnell** [comment]
@yuhanmao1207, I am happy to help!  Could you zip your inp/py file (or a cutdown version), and share here?  As for the feature, we don’t give specific access to that parameter. However, there are some quick ways to create the behavior in pure pyswmm.

**Reply by bemcdonnell** [reply]
@yuhanmao1207 it is really unclear to me what you are trying to accomplish.  Your notebook shows that you are using swmm_api to create your INP file in inject control rules therein.  Then you are running your model with pyswmm?   Are you asking for help building control rules in Python to control your model?  The reason I ask is in your PySWMM Simulation loop, you are not incorporating any control.  Can you walk through this example of writing control logic in pure python and see if this helps clarify what the pyswmm project is helping with?  This is the "espresso machine" example which helps you build control logic in python

https://www.pyswmm.org/examples#h.ndyju1ev0lg9

**Reply by yuhanmao1207** [reply]
Sorry for the misleading. I tried to use pyswmm, but I found there is no TIMEOPEN/TIMECLOSED feature. So I just tried swmm_api. But it's still not working for my case. And how can I control the pump by time in pure pyswmm? But anyway, thanks a lot for all the help. I will check it out.

:)
